src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -1294,35 +1533,49 @@ bool is_enum_superset(json::Value const& older, json::Value const& newer) {
         // also be improved with normalization of the input
         if (older_set.end() == std::ranges::find(older_set, v)) {
             // newer has an element not in older
-            return false;
+            res.emplace<json_incompatibility>(
+              std::move(enum_p), json_incompatibility_type::enum_array_changed);
+            return res;
         }
     }
 
-    return true;
+    return res;
 }
 
-bool is_not_combinator_superset(
-  context const& ctx, json::Value const& older, json::Value const& newer) {
+raw_compatibility_result is_not_combinator_superset(
+  context const& ctx,
+  json::Value const& older,
+  json::Value const& newer,
+  std::filesystem::path p) {
+    raw_compatibility_result res;
+
     auto older_it = older.FindMember("not");
     auto newer_it = newer.FindMember("not");
     auto older_has_not = older_it != older.MemberEnd();
     auto newer_has_not = newer_it != newer.MemberEnd();
 
     if (older_has_not != newer_has_not) {
         // only one has a "not" schema, not compatible
-        return false;
+        res.emplace<json_incompatibility>(
+          p, json_incompatibility_type::type_changed);
```

Comment:
Why isn't this `p / "not"`?

Reply:
That's what I would expect as well, but I think the reference implementation handles the "not" keyword slightly differently and all changes to "not" end up as either `type_changed` or `combined_type_changed`. See this test case and the note in the next one: https://github.com/redpanda-data/redpanda/blob/b29d510aa9c4adf52e552f82711ceec20a21ebbb/src/v/pandaproxy/schema_registry/test/test_json_schema.cc#L666-L679